### ISSUE TITLE
Issue with Bluebird.promisify()

### ISSUE THREAD

**Issue by cvburgess** [wildbit/postmark.js#5]
Howdy!

Tried promisifying the sendEmailBatch function and it gave me a peculiar error...

I tried using the callback style and it works, but promises are a must-have in my current project.

Here's the code:

``` js
const Promise = require('bluebird');
const postmark = require('postmark');

const client = new postmark.Client(<api_key>);

const messages = []; // array or message objects
const sendEmails = Promise.promisify(client.sendEmailBatch);

return sendEmails(messages);
```

```
Unhandled rejection TypeError: this.processRequestWithBody is not a function
at Client.sendEmailBatch (/<project path>/node_modules/postmark/lib/postmark/Client.js:120:14)
```

**Comment by cvburgess** [comment]
As a workaround for now, I am using this hack:

```js
return new Promise((resolve, reject) => {

    client.sendEmailBatch(messages, (error, success) => {

      if (error) reject(error);

      resolve(success);

    });

  });
```

**Comment by mansona** [comment]
Hey @cvburgess I have implemented something very similar using a corresponding function on [q.js](https://github.com/kriskowal/q) 

Essentially Q has two functions for wrapping node style functions: Q.denodeify and Q.nbind

Denodeify should be used for *static* functions and nbind should be used for functions **that are part of an object and make use of `this`**. In this case, you have to use Q.nbind with `postmark.sendEmail` because it is part of the postmark object. 

Here is my code: 
```
const postmark = require('postmark');
const Q = require('q');

const client = new postmark.Client(nconf.get('postmark:api_key'));
const sendEmail = Q.nbind(client.sendEmail, client);
```

Obviously, this doesn't directly apply to you but you will need to find the corresponding function to Q.nbind in Bluebird.

**Comment by atheken** [comment]
Hey folks, I just want to acknowledge this issue. I'll have a look at the code and see if it's an issue related to the prototype for `Client`.

**Comment by sudo-suhas** [comment]
I was able to work with bluebird by doing the following:

```js
const Promise = require('bluebird'), // eslint-disable-line no-shadow
    postmark = require('postmark');

const client = new postmark.Client(postmarkConfig.apiToken),
    sendEmailWithTemplate = Promise.promisify(client.sendEmailWithTemplate, {context: client});

// Usage
// Returns a promise
sendEmailWithTemplate({
    From: from,
    To: to,
    TemplateId: postmarkConfig.templates[type],
    TemplateModel: locals
});
```

**Comment by stringbeans** [comment]
as an FYI, this is not an issue with postmark.js but an issue with how you're using `Promise.promisify`. You must bind the correct context to the function you are "promisifying"

**Comment by cvburgess** [comment]
I will try this in my code and close this issue if it works - thanks @stringbeans :)

**Comment by atheken** [comment]
We just released `1.4.0` of the library, which includes `Promise` support. You to get Promises back, simply call the same client methods without a callback function. The client methods will return a Promise when the callback function is emitted.

**Comment by cvburgess** [comment]
Thank you @atheken I will update tonight :D
